### PR TITLE
Added all the nvSHMEM APIs to APINames_nvSHMEM.inc

### PR DESCRIPTION
**Pull request by TejaX-Alaghari**
Created the new APINames_nvSHMEM.inc file with all the nvSHMEM APIs and marked them as non-processed

### CODE REVIEW

File: clang/lib/DPCT/APINames_nvSHMEM.inc [hunk 1]
```diff
@@ -0,0 +1,2423 @@
...
+
+
+ENTRY(nvshmemx_buffer_register, nvshmemx_buffer_register, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_buffer_unregister, nvshmemx_buffer_unregister, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_buffer_unregister_all, nvshmemx_buffer_unregister_all, false, NO_FLAG, P4, "comment")
+
+
+ENTRY(nvshmem_team_my_pe, nvshmem_team_my_pe, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_team_n_pes, nvshmem_team_n_pes, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_team_config_t, nvshmem_team_config_t, false, NO_FLAG, P4, "comment")
```
Comment (tangjj11) [inline clang/lib/DPCT/APINames_nvSHMEM.inc]: `nvshmem_team_config_t` is not a API but a struct, it used in `nvshmem_team_get_config`.

File: clang/lib/DPCT/APINames_nvSHMEM.inc [hunk 2]
```diff
@@ -0,0 +1,2423 @@
...
+ENTRY(nvshmemx_get128_warp, nvshmemx_get128_warp, false, NO_FLAG, P4, "comment")
+
+
+ENTRY(nvshmem_getmem, nvshmem_getmem, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_getmem_on_stream, nvshmemx_getmem_on_stream, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_getmem_block, nvshmemx_getmem_block, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_getmem_warp, nvshmemx_getmem_warp, false, NO_FLAG, P4, "comment")
+
+
+ENTRY(nvshmem_float_p, nvshmem_float_p, false, NO_FLAG, P4, "comment")
```
Comment (tangjj11) [inline clang/lib/DPCT/APINames_nvSHMEM.inc]:
```suggestion
ENTRY(nvshmem_float_g, nvshmem_float_g, false, NO_FLAG, P4, "comment")
```

Comment (tangjj11) [reply clang/lib/DPCT/APINames_nvSHMEM.inc]: All the `_p` function should be chenged to `_g`.

File: clang/lib/DPCT/APINames_nvSHMEM.inc [hunk 3]
```diff
@@ -0,0 +1,2423 @@
...
+
+ENTRY(nvshmem_put8_nbi, nvshmem_put8_nbi, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_put16_nbi, nvshmem_put16_nbi, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_put32_nbi, nvshmem_put32_nbi, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_put64_nbi, nvshmem_put64_nbi, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_put128_nbi, nvshmem_put128_nbi, false, NO_FLAG, P4, "comment")
+
+ENTRY(nvshmemx_put8_nbi_on_stream, nvshmemx_put8_nbi_on_stream, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_put16_nbi_on_stream, nvshmemx_put16_nbi_on_stream, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_put32_signal_nbi_on_stream, nvshmemx_put32_signal_nbi_on_stream, false, NO_FLAG, P4, "comment")
```
Comment (tangjj11) [inline clang/lib/DPCT/APINames_nvSHMEM.inc]: Why only `put32` add a `signal`?

File: clang/lib/DPCT/APINames_nvSHMEM.inc [hunk 4]
```diff
@@ -0,0 +1,2423 @@
...
+ENTRY(nvshmemx_put64_nbi_on_stream, nvshmemx_put64_nbi_on_stream, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_put128_nbi_on_stream, nvshmemx_put128_nbi_on_stream, false, NO_FLAG, P4, "comment")
+
+ENTRY(nvshmemx_put8_nbi_block, nvshmemx_put8_nbi_block, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_put16_nbi_block, nvshmemx_put16_nbi_block, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_put32_signal_nbi_block, nvshmemx_put32_signal_nbi_block, false, NO_FLAG, P4, "comment")
```
Comment (tangjj11) [inline clang/lib/DPCT/APINames_nvSHMEM.inc]: Why only put32 add a signal?

File: clang/lib/DPCT/APINames_nvSHMEM.inc [hunk 5]
```diff
@@ -0,0 +1,2423 @@
...
+ENTRY(nvshmemx_put64_nbi_block, nvshmemx_put64_nbi_block, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_put128_nbi_block, nvshmemx_put128_nbi_block, false, NO_FLAG, P4, "comment")
+
+ENTRY(nvshmemx_put8_nbi_warp, nvshmemx_put8_nbi_warp, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_put16_nbi_warp, nvshmemx_put16_nbi_warp, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmemx_put32_signal_nbi_warp, nvshmemx_put32_signal_nbi_warp, false, NO_FLAG, P4, "comment")
```
Comment (tangjj11) [inline clang/lib/DPCT/APINames_nvSHMEM.inc]: Why only put32 add a signal?

File: clang/lib/DPCT/APINames_nvSHMEM.inc [hunk 6]
```diff
@@ -0,0 +1,2423 @@
...
+ENTRY(nvshmem_uint64_atomic_or, nvshmem_uint64_atomic_or, false, NO_FLAG, P4, "comment")
+
+ENTRY(nvshmem_uint_atomic_fetch_xor, nvshmem_uint_atomic_fetch_xor, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_ulong_atomic_fetch_xor, nvshmem_ulong_atomic_fetch_xor, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_ulonglong_atomic_fetch_xor, nvshmem_ulonglong_atomic_fetch_xor, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_int32_atomic_fetch_xor, nvshmem_int32_atomic_fetch_xor, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_int64_atomic_fetch_xor, nvshmem_int64_atomic_fetch_xor, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_uint32_atomic_fetch_xor, nvshmem_uint32_atomic_fetch_xor, false, NO_FLAG, P4, "comment")
+ENTRY(nvshmem_uint64_atomic_fetch_xor, nvshmem_uint64_atomic_fetch_xor, false, NO_FLAG, P4, "comment")
+
```
Comment (tangjj11) [inline clang/lib/DPCT/APINames_nvSHMEM.inc:972]: Miss some `xxx_atomic_xor` function.

**Review by tangjj11 — CHANGES_REQUESTED** [review]

File: clang/lib/DPCT/APINames_nvSHMEM.inc (hunk 1 above)
Comment (TejaX-Alaghari) [reply clang/lib/DPCT/APINames_nvSHMEM.inc]: Removed the entry

File: clang/lib/DPCT/APINames_nvSHMEM.inc (hunk 3 above)
Comment (TejaX-Alaghari) [reply clang/lib/DPCT/APINames_nvSHMEM.inc]: It was a misplaced replacement
Corrected it now

File: clang/lib/DPCT/APINames_nvSHMEM.inc (hunk 4 above)
Comment (TejaX-Alaghari) [reply clang/lib/DPCT/APINames_nvSHMEM.inc]: It was a misplaced replacement
Corrected it now

File: clang/lib/DPCT/APINames_nvSHMEM.inc (hunk 5 above)
Comment (TejaX-Alaghari) [reply clang/lib/DPCT/APINames_nvSHMEM.inc]: It was a misplaced replacement
Corrected it now

File: clang/lib/DPCT/APINames_nvSHMEM.inc (hunk 2 above)
Comment (TejaX-Alaghari) [reply clang/lib/DPCT/APINames_nvSHMEM.inc]: nvshmem_TYPENAME_p APis exiist [here](https://docs.nvidia.com/nvshmem/api/gen/api/rma.html#nvshmem-p)

File: clang/lib/DPCT/APINames_nvSHMEM.inc (hunk 6 above)
Comment (TejaX-Alaghari) [reply clang/lib/DPCT/APINames_nvSHMEM.inc:972]: Added the APIs

File: clang/lib/DPCT/APINames_nvSHMEM.inc (hunk 2 above)
Comment (tangjj11) [reply clang/lib/DPCT/APINames_nvSHMEM.inc]: no, I mean the _p APIs has been added in line 210, this section you need add the _g APIs but not _p.

**Review by tangjj11 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**